tests/test_mutations.py

```diff
@@ -936,6 +937,91 @@ def verify_model_ts(self, ts, model, discrete, verify_roots):
             p /= sum(p)
             self.chisquare(row, sum(row) * p)
 
+    def verify_model_ts_general(self, ts, model, discrete, verify_roots, mutation_rate):
+        alleles = [a.decode() for a in model.alleles]
+        num_alleles = len(alleles)
+        roots = np.zeros((num_alleles,))
+
+        if discrete:
+            Q = model.transition_matrix.copy()
+            np.fill_diagonal(Q, 0.0)
+            row_sums = Q.sum(axis=1)
+            np.fill_diagonal(Q, -1 * row_sums)
+            observed_transitions = np.zeros((num_alleles, num_alleles))
+            expected = np.zeros((num_alleles, num_alleles))
+
+            for v in ts.variants(samples=range(ts.num_nodes), impute_missing_data=True):
+                t = ts.at(v.position)
+                for n in t.nodes():
+                    pn = t.parent(n)
+                    if pn != tskit.NULL:
+                        pa = v.alleles[v.genotypes[pn]]
+                        da = v.alleles[v.genotypes[n]]
+                        observed_transitions[alleles.index(pa), alleles.index(da)] += 1
+                        expected += _matrix_exponential(
+                            t.branch_length(n) * mutation_rate * Q
+                        )
```

Comment:
whoops - this is the expected number *conditioned* on the parent of the branch, so I think it should be
```
j = alleles.index(pa)
expected[, :] += _matrix_exponential(
    t.branch_length(n) * mutation_rate * Q)[j, :]
```

Reply:
Oh, wait - this is a bit trickier, because the observed numbers are conditioned on having at least one mutation on the tree. Hm.  I don't think what you had is right, because since *all* rows of exp(t mu Q) are included, that's essentially assuming the uniform distribution over alleles at the head of every branch.  But, maybe I'm missing something? I'm having a hard time seeing what the right way to compute this is.